application/controllers/SurveyAdministrationController.php

```diff
@@ -2350,10 +2361,16 @@ public function actionExpireMultipleSurveys()
             $survey = Survey::model()->findByPk($sid);
             $survey->expires = $expires;
             $aResults[$survey->primaryKey]['title'] = ellipsize($survey->correct_relation_defaultlanguage->surveyls_title, 30);
-            if ($survey->save()) {
-                $aResults[$survey->primaryKey]['result'] = true;
-            } else {
+            if (!Permission::model()->hasGlobalPermission('superadmin')) {
                 $aResults[$survey->primaryKey]['result'] = false;
+                $aResults[$survey->primaryKey]['error'] = gT("User does not have valid permissions");
+            } else {
+                if ($survey->save()) {
+                    $aResults[$survey->primaryKey]['result'] = true;
+                } else {
+                    $aResults[$survey->primaryKey]['result'] = false;
+                    $aResults[$survey->primaryKey]['error'] = gT("Survey update failed");
```

Comment:
Maybe $survey->getErrors() ?

Reply:
I don't think those errors are descriptive, but too technical. If there would be a log or an "advanced results" section, would be good idea. But here, not sure if it is a good fit. We are not doing that anywhere else I think.

Reply:
`errorSummary($osSurvey)`, not sure how this could be used when the output is for multiple surveys. Maybe too much information for the grid. I would say to leave it for another ticket?

Reply:
I put a **?**

OK for stay (must check what error can be done ;).

Maybe log it then ?